trinity/trainer/trainer.py

```diff
@@ -119,6 +120,14 @@ def flush_log(self, step: int) -> None:
         """Flush the log of the current step."""
         self.engine.logger.log({}, step=step, commit=True)
 
+    def shutdown(self) -> None:
+        # if checkpoint not saved, save the last checkpoint
+        step_num = self.engine.global_steps
+        path = os.path.join(self.config.model.checkpoint_path, f"global_step_{step_num}")
```

Comment:
Check the current step and save only if there is no checkpoint saved in the current step